proxy/src/auth/backend/console.rs

```diff
@@ -139,16 +148,21 @@ impl<'a> Api<'a> {
     }
 
     async fn get_auth_info(&self) -> Result<AuthInfo, GetAuthInfoError> {
-        let mut url = self.endpoint.clone();
-        url.path_segments_mut().push("proxy_get_role_secret");
-        url.query_pairs_mut()
-            .append_pair("project", self.creds.project().expect("impossible"))
-            .append_pair("role", self.creds.user);
+        let req = self
+            .endpoint
+            .get("proxy_get_role_secret")
+            .header("X-Request-ID", self.extra.session_id)
+            .query(&[
+                ("application_name", self.extra.application_name),
+                ("project", Some(self.creds.project().expect("impossible"))),
+                ("role", Some(self.creds.user)),
+            ])
+            .build()?;
```

Comment:
@ololobus do you think this makes sense? I'm specifically interested in your opinion on `X-Request-ID` & `application_name`. See a similar request below in the same file.

Reply:
Strong +1 for request id as console already can digest it and add into context. Not sure that it's properly used everywhere after that, though.

Yet, if it's absent console generated new request id like this: https://github.com/neondatabase/cloud/blob/fa7a5db09a24c5d519bbd75f7ca723dc7bdcd822/goapp/internal/requestid/requestid.go#L19

So I wonder, would it be better if proxy will pass `X-Request-ID` in the same format for consistency? Or we can agree on some different format, but use it everywhere

Reply:
Although, passing `session_id` seems to be important too, as we do two requests per session: get secret and wake up

Reply:
Ah, missed this part, sorry. Then I even more confused as I thought that `session_id` is what is shown to the user, when one does `psql -h connect.neon.tech`

So this `session_id` passed as `X-Request-ID` in this PR has nothing common with internal proxy session id?

Reply:
It is nice to have a unique Request ID per every outgoing HTTP request to the console. If we reuse Request ID for different requests and one of them has failed, then one who got a Request ID of a failed request from proxy logs, will not have the ability grep console logs by this request id and differ one request from another. IIUC `self.extra.session_id` is used as a Request ID for different HTTP requests. Can we change that to a unique value per request?

Reply:
> So this session_id passed as X-Request-ID in this PR has nothing common with internal proxy session id?

To be clear, there's no such thing as an "internal proxy session id". What we have here is a per-client-connection uuid that's used one time for either one or two http requests. When we accept a new connection, we query auth info and then wake up the node, hence 1 or 2 calls. Mind you, those requests are different, there's never a case where you use the same id for the same kind of request, so the problem you're talking about is impossible.

I call it a session id because that's what it is, but that doesn't mean that we'll use this for 2 or more wakeup calls.

Reply:
Alternatively, we could do as suggested earlier by @ololobus: pass two IDs: one unique (that's `X-Request-ID`) and the other one (idk, `X-Session-ID`?) as it is now.

Reply:
Yep, I think that two ids are better:
- unique `request_id` per each http request to the control-plane
- unique `session_id` per each client session

Ideally, that's what I want to see in the proxy logs:
```
INFO: new client connection, session_id: xxx1
INFO: control-plane get_auth request finished with 200 OK, session_id: xxx1, request_id: yyy1
INFO: control-plane wakeup request finished with 200 OK, session_id: xxx1, request_id: yyy2
INFO: client connected to the compute node, session_id: xxx1
INFO: new client connection, session_id: xxx2
ERROR: control-plane get_auth request finished with 500 InternalServerError, session_id: xxx2, request_id: yyy3, error: I'm tired
ERROR: client connection attempt failed, session: xxx2
```

That way, we can transparently track any particular connection attempt **AND** any particular control-plane request. And this `session_id` should be proxy internal id for the session and it should be logged for each log record corresponding to this session. Name `session_id` totally makes sense to me, BTW.

Does it make sense to you? If yes and you'll show me the issue for proxy logging you mentioned above, I can copy-paste it there :)

--

Sorry, I'm stupid, but what is this **565752694af9abf1** in the `https://console.stage.neon.tech/psql_session/565752694af9abf1`? Is it `session_id`? Does it have the same format for scram-proxy?

Reply:
> Yep, I think that two ids are better:
> * unique request_id per each http request to the control-plane
> * unique session_id per each client session

Fine, let's do this :)

> And this session_id should be proxy internal id for the session and it should be logged for each log record corresponding to this session. 

That's exactly the plan, I just don't want to do everything in one PR, because that'd be much more code.

> Sorry, I'm stupid, but what is this 565752694af9abf1 in the https://console.stage.neon.tech/psql_session/565752694af9abf1? Is it session_id? Does it have the same format for scram-proxy?

No, it's just a random `hex(u64)`. `scram-proxy` doesn't use that, in fact, we have 3 auth backends: console (uses scram exclusively), postgres (to debug scram locally) and link (which uses this u64). This leads me to think that our current naming is not as good is it could be: we should've named those instances `proxy-link` and `proxy-console`.